src/document_basic.c

```diff
@@ -178,63 +178,68 @@ int Document_LoadAllFields(Document *doc, RedisModuleCtx *ctx) {
   return rc;
 }
 
+typedef struct {
+    RedisModuleCtx *ctx;
+    size_t nreplies;
+    size_t lang_len;
+    size_t score_len;
+    size_t payload_len;
+    char* lang_field;
+    char* score_field;
+    char* payload_field;
+} Document_ReplyAllFields_privdata;
+
+static void Document_ReplyAllFields_scan_callback(RedisModuleKey *key, RedisModuleString* field, RedisModuleString* value, void *privdata) {
+    REDISMODULE_NOT_USED(key);
+    Document_ReplyAllFields_privdata* pd = privdata;
+    size_t fieldCStrLen;
+    const char* fieldCStr = RedisModule_StringPtrLen(field, &fieldCStrLen);
+    RS_LOG_ASSERT(fieldCStrLen > 0, "field string cannot be empty");
+    if ((pd->lang_len == fieldCStrLen && strncasecmp(fieldCStr, pd->lang_field, fieldCStrLen) == 0) ||
+        (pd->score_len == fieldCStrLen && strncasecmp(fieldCStr, pd->score_field, fieldCStrLen) == 0) ||
+        (pd->payload_len == fieldCStrLen && strncasecmp(fieldCStr, pd->payload_field, fieldCStrLen) == 0)) {
+      return;
+    }
+    RedisModule_ReplyWithStringBuffer(pd->ctx, fieldCStr, fieldCStrLen);
+    if(value){
+        size_t valueCStrLen;
+        const char* valueCStr = RedisModule_StringPtrLen(value, &valueCStrLen);
+        RedisModule_ReplyWithStringBuffer(pd->ctx, valueCStr, valueCStrLen);
+    } else {
+        RedisModule_ReplyWithNull(pd->ctx);
+    }
+    pd->nreplies++;
+}
+
+
 int Document_ReplyAllFields(RedisModuleCtx *ctx, IndexSpec *spec, RedisModuleString *id) {
   int rc = REDISMODULE_ERR;
   RedisModuleCallReply *rep = NULL;
-
```

Comment:
This is used by ft.get and ft.mget which are deprecated, so I do not see a need to invest in optimizing it.

Reply:
Reverted changes on Document_ReplyAllFields() :)